app/src/main/java/org/oppia/android/app/drawer/NavigationDrawerFragmentPresenter.kt

```diff
@@ -263,14 +266,34 @@ class NavigationDrawerFragmentPresenter @Inject constructor(
     )
   }
 
-  fun markHomeMenuCloseDrawer() {
-    binding.fragmentDrawerNavView.menu.getItem(
-      NavigationDrawerItem.HOME.ordinal
-    ).isChecked =
-      true
+  fun restoreLastCheckedMenuItem() {
+    if (previousMenuItemId != null) {
+      val lastCheckedMenuItemValue: Int = previousMenuItemId!!
+      if (lastCheckedMenuItemValue == 0) {
+        getFooterViewModel().isAdministratorControlsSelected.set(true)
+        uncheckAllMenuItemsWhenAdministratorControlsIsSelected()
+        binding.fragmentDrawerNavView.menu.getItem(
+          NavigationDrawerItem.SWITCH_PROFILE.ordinal
+        ).isChecked =
+          false
+      } else {
+        binding.fragmentDrawerNavView.menu.getItem(
+          NavigationDrawerItem.valueFromNavId(
+            lastCheckedMenuItemValue
+          ).ordinal
+        ).isChecked =
+          true
+      }
+    }
```

Comment:
```suggestion
    previousMenuItemId?.let { itemId ->
      if (itemId == 0) {
        getFooterViewModel().isAdministratorControlsSelected.set(true)
        uncheckAllMenuItemsWhenAdministratorControlsIsSelected()
        binding.fragmentDrawerNavView.menu.getItem(
          NavigationDrawerItem.SWITCH_PROFILE.ordinal
        ).isChecked =
          false
      } else {
        binding.fragmentDrawerNavView.menu.getItem(
          NavigationDrawerItem.valueFromNavId(itemId).ordinal
        ).isChecked = true
      }
    }
```

More Kotlin-idiomatic.

Reply:
Done.